.github/workflows/CI-pr.yml

```diff
@@ -0,0 +1,135 @@
+name: CI-pr
+
+on:
+  pull_request:
+    types:
+      - opened
+      - ready_for_review
+      - reopened
+      - synchronize
+
+jobs:
+  formatting:
+    name: Check formatting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate a token
```

Comment:
This step would have to be run only for PRs from the current repository, can't run it on PRs from forks.